ports/esp32/espneopixel.c

```diff
@@ -24,14 +24,14 @@ void IRAM_ATTR esp_neopixel_write(uint8_t pin, uint8_t *pixels, uint32_t numByte
 
     if (timing == 1) {
         // 800 KHz
-        time0 = (fcpu * 0.35) / 1000000; // 0.35us
-        time1 = (fcpu * 0.8) / 1000000; // 0.8us
-        period = (fcpu * 1.25) / 1000000; // 1.25us per bit
+        time0 = (uint32_t)((fcpu * 0.35) / 1000000.); // 0.35us
+        time1 = (uint32_t)((fcpu * 0.8) / 1000000.); // 0.8us
+        period = (uint32_t)((fcpu * 1.25) / 1000000.); // 1.25us per bit
     } else {
         // 400 KHz
-        time0 = (fcpu * 0.5) / 1000000; // 0.35us
-        time1 = (fcpu * 1.2) / 1000000; // 0.8us
-        period = (fcpu * 2.5) / 1000000; // 1.25us per bit
+        time0 = (uint32_t)((fcpu * 0.5) / 1000000.); // 0.35us
+        time1 = (uint32_t)((fcpu * 1.2) / 1000000.); // 0.8us
+        period = (uint32_t)((fcpu * 2.5) / 1000000.); // 1.25us per bit
```

Comment:
This code should really be rewritten to use plain ints (it's easy, eg `time0 = fcpu * 350 / 1000`, with `fcpu = ets_get_cpu_frequency()`), but that's out of scope of this commit.  I'd be happy to make this change separately, but it'd need to go before this PR else the build would error.

Reply:
Good point, yes please go ahead and change it already.

Reply:
See #5888.  I'll need to test then can merge that.

Reply:
Ok, that's now merged.